### PR TITLE
Add domain type for `max_msg_num` and related types and a fix for #1971

### PR DESCRIPTION
**Pull request by romac**
Closes: #1971

______

### PR author checklist:
- [x] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [x] Added tests: integration (for Hermes) or unit/mock tests (for modules). 
- [x] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

**Comment by adizere** [comment]
Please see the work in https://github.com/informalsystems/ibc-rs/pull/1980/files for reproducing the error.

> Soares: Consider copying the test in this branch to repro the error.

**Comment by adizere** [comment]
Please see the work in https://github.com/informalsystems/ibc-rs/pull/1980/files for reproducing the error.

> Soares: Consider copying the test in this branch to repro the error.

**Comment by romac** [comment]
Not sure if that makes sense now that we cannot set `chain_config.max_msg_num` to 1 anymore. On the other hand I can include a config parsing test.

**Review by adizere — APPROVED** [review]
Great to see the unit tests and domain types here, neat!

**Comment by ancazamfir** [comment]
Can we hold on here until we clarify a bit?

**Review by ancazamfir — CHANGES_REQUESTED** [review]
Can we hold off on this one a bit? see https://github.com/informalsystems/ibc-rs/issues/1971#issuecomment-1116022780

File: relayer/src/config/types.rs
Comment (romac) [reply relayer/src/config/types.rs:258]: No, good catch!

Comment (romac) [reply relayer/src/config/types.rs]: Nevermind we do need it, added a comment

File: relayer/src/error.rs
Comment (seanchen1991) [inline relayer/src/error.rs:529]: Is there any reason to leave this old logic in the code as a comment? Would the SDK potentially move back to returning the old error code?

Comment (romac) [reply relayer/src/error.rs:529]: I think it's quite nice to leave some documentation as to why we are not checking the status code like we do in the other places. I would vote for keeping it in.

File: .changelog/unreleased/improvements/ibc-relayer/1971-max-msg-num-min-bound.md
Comment (adizere) [inline .changelog/unreleased/improvements/ibc-relayer/1971-max-msg-num-min-bound.md:1]: Let's delete this file before merging.

**Review by adizere — APPROVED** [review]

File: .changelog/unreleased/improvements/ibc-relayer/1971-max-msg-num-min-bound.md
Comment (romac) [reply .changelog/unreleased/improvements/ibc-relayer/1971-max-msg-num-min-bound.md:1]: Isn't that a change worth documenting?

Comment (adizere) [reply .changelog/unreleased/improvements/ibc-relayer/1971-max-msg-num-min-bound.md:1]: But we're not enforcing it to be bigger than 2 anymore. From a user perspective, not much changes.

I guess we can keep the changelog, but then let's adjust the message to be consistent with the implementation. As I understand it, we allow it to be [1, 100] with a default of 30.

https://github.com/informalsystems/ibc-rs/blob/143e7f3dc16b568acb7ade342b05febe1d84183a/relayer/src/config/types.rs#L31-L33

Comment (romac) [reply .changelog/unreleased/improvements/ibc-relayer/1971-max-msg-num-min-bound.md:1]: Right, my bad!

Comment (romac) [reply .changelog/unreleased/improvements/ibc-relayer/1971-max-msg-num-min-bound.md]: Fixed

**Pull request merged**
